crates/forge/bin/cmd/install.rs

```diff
@@ -123,12 +123,31 @@ impl DependencyInstallOpts {
         let install_lib_dir = config.install_lib_dir();
         let libs = git.root.join(install_lib_dir);
 
-        if dependencies.is_empty() && !self.no_git {
+        // Pre-emptively create the directory where deps will be installed if it's missing
+        fs::create_dir_all(&libs)?;
+
+        // Only update submodule deps if there are no other deps to install, we're using git
+        // and an appropiate .gitmodules file exists.
+        if dependencies.is_empty() && !self.no_git && git.root.join(".gitmodules").exists() {
             p_println!(!self.quiet => "Updating dependencies in {}", libs.display());
-            // recursively fetch all submodules (without fetching latest)
-            git.submodule_update(false, false, false, true, Some(&libs))?;
+
+            let empty_install_dir = libs.read_dir()?.next().is_none();
+
+            if !empty_install_dir {
+                fs::create_file(libs.join(".keep"))?;
+                // Update the git project with the newly created folder if we're using git.
+                git.add(Some(&libs.join(".keep")))?;
```

Comment:
this looks super hacky,

I'm still skeptical that any of this is necessary.

The edge you're described is correct, but this can only happen if the lib dir is manually removed but already known to git.

so what we really want to know here is "does this repo contain any submodules"
I think the right fix here is querying `git submodule status` first
if this doesn't give us submodules, then there's nothing to update

Reply:
impl'd in https://github.com/foundry-rs/foundry/pull/6521/commits/84a83a95fd507a722f0446f97c892ec09a5d2a9e